### PR TITLE
Feature/tv 10007 update node sdk errors

### PR DESCRIPTION
**Pull request by mscherer11**
Removes errors from api method calls

### CODE REVIEW

File: src/lib/tv-request.js [hunk 1]
```diff
@@ -41,14 +41,18 @@ class tvRequest {
...
 
-                if(isErrorResponse(res.statusCode)) {
-                    reject(res.statusCode)
-                }
+              if (err) {
+                return reject(err);
+              }
+
+              if(isErrorResponse(res.statusCode)) {
+                  reject(res);
```
Comment (jetherton) [inline src/lib/tv-request.js]: You don't have a return here, but you do on line 50. Which is the right way to do this? @slinhart any thoughts on this.

File: src/trackvia-api.js [hunk 2]
```diff
@@ -78,14 +72,7 @@ class TrackviaAPI {
         if(!name) {
             throw new Error('Must provide name argument for getApp');
```
Comment (jetherton) [inline src/trackvia-api.js]: Should this be refactored too? so all errors go through the same route?

File: test/runner.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+var TrackviaAPI = require('../src/trackvia-api');
+
+var tvApi = new TrackviaAPI('06e44182a2304049fa8fab34251d8db5', 'https://go.trackvia.com')
+
+function testIt() {
+  tvApi.login('michael.c.scherer@gmail.com', 'WaterBottleEngine1')
```
Comment (jetherton) [inline test/runner.js]: Lets not use real passwords / account keys for things we check in. To work around this in a sustainable way, I'd recommend creating a test/config.js.template and check that in with dummy data for username/password/key, then make the test run with a test/config.js file that's also added in .gitignore so you can't check it in by accident.

File: src/lib/tv-request.js (hunk 1 above)
Comment (slinhart) [inline src/lib/tv-request.js]: almost always, you want to return when you reject.

**Pull request merged**